chain/arweave/src/chain.rs

```diff
@@ -44,20 +44,20 @@ impl std::fmt::Debug for Chain {
     }
 }
 
-impl Chain {
-    pub fn new(
+impl BuildableBlockchain for Chain {
+    fn build(
         logger_factory: LoggerFactory,
-        name: String,
+        chain_id: String,
         chain_store: Arc<dyn ChainStore>,
-        firehose_endpoints: FirehoseEndpoints,
-        metrics_registry: Arc<dyn MetricsRegistry>,
+        fh_endpoints: FirehoseEndpoints,
```

Comment:
please leave this one as was, I have a PR that will replace FIrehoseEndpoints with ChainClient in most places